requirements/ci/py38.yml

```diff
@@ -10,7 +10,7 @@ dependencies:
   - setuptools >=40.8.0
 
 # Core dependencies.
-  - cartopy >=0.18
+  - cartopy >=0.20
```

Comment:
I think this is probably the right thing to do, but worth considering the ramifications for other dependencies we have.  

I think that cartopy 0.20 adopts [NEP29](https://numpy.org/neps/nep-0029-deprecation_policy.html), which will, I assume result in us also adopting NEP29 on dependencies iris shares with cartopy (numpy, matplotlib, etc)

Reply:
Maybe iris embracing NEP29 is something to discuss at a peloton?

Reply:
There's been chatter about also explicitly adopting NEP29 akin to `cartopy`. I'd happily discuss that if anyone is keen.

Python versions and releases a quite predictable, given the level of planning that is involved to schedule and announce it's releases to the community... but that's a wee bit harder/impossible for other dependencies. However, Python is the common factor here, and naturally might introduce "healthy" minimum pins.

We could discuss it at a peloton, but do you fancy also spinning up a dedicated GH Discussion? That way others who don't attend a peloton can have their say and get involved

Reply:
@jamesp Awesome, thanks for creating the [GH Discussion on NEP29](https://github.com/SciTools/iris/discussions/4332) :beers: